miio/gateway.py

```diff
@@ -744,6 +802,21 @@ def get_firmware_version(self) -> Optional[int]:
             )
         return self._fw_ver
 
+    @command()
+    def uninstall_scripts(self, encoded_token):
+        return dict(
+            map(
+                lambda action: (
+                    action,
+                    (
+                        action_id[action](self.sid),
+                        self._gw.x_del(action_id[action](self.sid)),
+                    ),
+                ),
+                action_id.keys(),
+            )
+        )
```

Comment:
This is way coo complicated and rather hard to interpret.

Is the goal to remove all scripts? Is there a way to get a list of current events, that could be iterated and then removed? How does removal/overwriting of a single script look like?

How is the encoded token used?

Reply:
yes... goal is to remove all scripts at least potentially could be installed in device... As far as i get list of installed scripts is not exposed through miio protocol... Here I xdel all scripts for all possible scripts to avoid memory leak on gw

Encoded token is used during script installation and somehow linked with real token. Unfortunately algo is unknown...

Reply:
Actually, I think there were some sort of list functionality related to scenes in the gateway implementation at some point, that could be related to this?

Anyway, I pointed out the encoded_token here as it is given as a parameter but never used in the method itself?